frontend/talentsearch/resources/js/components/workLocationPreferenceForm/WorkLocationPreferenceForm.tsx

```diff
@@ -0,0 +1,173 @@
+    [WorkRegion.North]: {
+      defaultMessage: "North Region: Yukon, Northwest Territories and Nunavut.",
+      description: "The work region of Canada described as North.",
+    },
+  });
+
+  const getWorkPreferenceregion = (
+    workRegionId: string | number,
+  ): MessageDescriptor =>
+    getOrThrowError(
+      workRegionsForWorkPreferenceForm,
+      workRegionId,
+      `Invalid Work Region '${workRegionId}'`,
+    );
+
+  return (
+    <ProfileFormWrapper
+      description="Indicate all locations where you are willing to work, including your current location (if you are interested in working there)."
+      title="Work location"
+      crumbs={[
+        {
+          title: intl.formatMessage({
+            defaultMessage: "Work Location Preference",
+            description:
+              "Display Text for the current page in Work Location Preference Form Page",
+          }),
+        },
+      ]}
+    >
+      <BasicForm onSubmit={onSubmit}>
+        <div>
+          <div data-h2-flex-item="b(1of1)" data-h2-padding="b(top, m)">
+            <div data-h2-padding="b(right, l)" data-testid="workLocation">
+              <Checklist
+                idPrefix="work-location"
+                legend={intl.formatMessage({
+                  defaultMessage: "Work location",
+                  description:
+                    "Legend for optional work preferences check list in work preferences form",
+                })}
+                name="workLocations"
+                items={enumToOptions(WorkRegion).map(({ value }) => ({
+                  value,
+                  label: intl.formatMessage(getWorkPreferenceregion(value)),
+                }))}
+                rules={{ required: intl.formatMessage(errorMessages.required) }}
+              />
+            </div>
+          </div>
+          <div data-h2-flex-item="b(1of1)" data-h2-padding="b(top, m)">
+            <div data-h2-padding="b(right, l)">
+              <p>
+                Indicate if there is a city that you would like to exclude from
+                a region.
+              </p>
+              <p data-h2-font-color="b([dark]gray)">
+                E.g.: You want to be considered for the Quebec region, but not
+                for Montréal.
+              </p>
+            </div>
+          </div>
+          <div data-h2-flex-item="b(1of2)" data-h2-padding="b(top, m)">
+            <div data-h2-padding="b(right, l)">
+              <TextArea
+                id="location-exemptions"
+                label="Location exemptions"
+                name="locationExemption"
+                placeholder="Optionally, add a city or village here..."
+              />
+            </div>
+          </div>
+        </div>
+        <ProfileFormFooter mode="saveButton" />
+      </BasicForm>
+    </ProfileFormWrapper>
+  );
+};
+export const WorkLocationPreferenceApi: React.FunctionComponent = () => {
+  const [, executeMutation] = useCreateWorkLocationPreferenceMutation();
```

Comment:
We need another query operation to get initial data as well.

Reply:
Done! Check again please!